pkg/cloudprovider/providers/gce/kms.go

```diff
@@ -112,15 +118,21 @@ func (gce *GCECloud) getGCPCloudKMSService(config io.Reader) (envelope.Service,
 	if err != nil && unrecoverableCreationError(err) {
 		return nil, err
 	}
+
 	parentName = parentName + "/cryptoKeys/" + cryptoKey
 
+	hasher := sha1.New()
+	hasher.Write([]byte(parentName))
+
 	service := &gkmsService{
 		parentName:      parentName,
+		parentHash:      base64.URLEncoding.EncodeToString(hasher.Sum(nil)),
 		cloudkmsService: cloudkmsService,
 	}
 
 	// Sanity check before startup. For non-GCP clusters, the user's account may not have permissions to create
 	// the key. We need to verify the existence of the key before apiserver startup.
+	// This also sets the latestKeyVersion known to this transformer.
 	_, err = service.Encrypt([]byte("test"))
 	if err != nil {
 		return nil, fmt.Errorf("failed to encrypt data using Google cloudkms, using key %s. Ensure that the keyRing and cryptoKey exist. Got error: %v", parentName, err)
```

Comment:
Here we say key to the user, but the code says parentName. I couldn't see the usage of parent name in the KMS docs. It seems like:

parentName = keyName
key = keyVersion 

I'd prefer the naming on the right hand side.

Reply:
Sorry for the delayed response.
I borrowed the term `parentName` from the official examples for encryption/decryption:
https://cloud.google.com/kms/docs/encrypt-decrypt#kms-howto-encrypt-go

The key -> keyVersion change sounds good though.